src/main/java/Entities/User.java

```diff
@@ -163,6 +163,7 @@ public void setCourses(Courses newCourses) {
     public void setInterests(HashMap<Integer, Boolean> newInterests) {
         this.interests = new Interests();
         this.interests.setData(newInterests);
+        this.interests.data = newInterests;
```

Comment:
This line seems redundant. Not sure what this is intended to do that is different from line 165.

Reply:
Facts